### PR TITLE
#197 - Attachment points should be within R-Group brackets

### PR DESCRIPTION
**Pull request by StarlaStarla**
Closing #197 
## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)
including the attachment points bounding box as accumulating the r-group box

## Check list
- [ ] unit-tests written
- [ ] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct

### CODE REVIEW

File: packages/ketcher-core/src/application/render/restruct/rergroup.js [hunk 1]
```diff
@@ -63,19 +63,34 @@ class ReRGroup extends ReObject {
 
     if (ret) ret = ret.extend(BORDER_EXT, BORDER_EXT)
 
+    let attachmentPointsVBox = render.ctab.getAttachmentsPointsVBox(
```
Comment (Nitvex) [inline packages/ketcher-core/src/application/render/restruct/rergroup.js:66]: If the variable doesn't change, please use `const` instead

File: packages/ketcher-core/src/application/render/restruct/rergroup.js [hunk 2]
```diff
@@ -63,19 +63,34 @@ class ReRGroup extends ReObject {
...
+      this.getAtoms(render)
+    )
+    if (attachmentPointsVBox)
```
Comment (Nitvex) [inline packages/ketcher-core/src/application/render/restruct/rergroup.js:69]: Could you please add `{ }` for even single-line `if`?

File: packages/ketcher-core/src/application/render/restruct/rergroup.js [hunk 3]
```diff
@@ -63,19 +63,34 @@ class ReRGroup extends ReObject {
...
+      attachmentPointsVBox = attachmentPointsVBox.extend(BORDER_EXT, BORDER_EXT)
+
+    ret =
```
Comment (Nitvex) [inline packages/ketcher-core/src/application/render/restruct/rergroup.js:72]: If you understood what `ret` means, I would like to ask you to rename it

File: packages/ketcher-core/src/application/render/restruct/rergroup.js [hunk 4]
```diff
@@ -63,19 +63,34 @@ class ReRGroup extends ReObject {
...
+    let bb = this.calcBBox(render)
 
     if (!bb) {
       console.error(
         'Abnormal situation, empty fragments must be destroyed by tools'
       )
       return {}
+    } else {
+      // add a little space between the attachment points and brackets
+      const vext = new Vec2(0.2, 0.4)
```
Comment (Nitvex) [inline packages/ketcher-core/src/application/render/restruct/rergroup.js:92]: And here :) 
Unfortunately, there are a lot of variables with meaningless name. Let's not add new ones.

File: packages/ketcher-core/src/application/render/restruct/rergroup.js (hunk 1 above)
Comment (StarlaStarla) [reply packages/ketcher-core/src/application/render/restruct/rergroup.js:66]: but I did change the value at line 70

File: packages/ketcher-core/src/application/render/restruct/rergroup.js (hunk 2 above)
Comment (StarlaStarla) [reply packages/ketcher-core/src/application/render/restruct/rergroup.js:69]: sure, will do

File: packages/ketcher-core/src/application/render/restruct/rergroup.js (hunk 3 above)
Comment (StarlaStarla) [reply packages/ketcher-core/src/application/render/restruct/rergroup.js:72]: sure, will do

File: packages/ketcher-core/src/application/render/restruct/rergroup.js (hunk 4 above)
Comment (StarlaStarla) [reply packages/ketcher-core/src/application/render/restruct/rergroup.js:92]: sure, will do

**Comment by Zhirnoff** [comment]
Made only for R-Group Fragment but not for S-Group
2023-07-07_13h21_47
2023-07-07_13h21_56

**Review by rrodionov91 — APPROVED** [review]

**Comment by StarlaStarla** [comment]
The S-Group fix please refer to this bug  https://github.com/epam/ketcher/issues/2949

**Pull request merged**